Python/Product/Analysis/Infrastructure/PathUtils.cs

```diff
@@ -318,5 +318,87 @@ static class PathUtils {
             // likely nothing we can do.
             return !Directory.Exists(path);
         }
+
+        public static FileStream OpenWithRetry(string file, FileMode mode, FileAccess access, FileShare share) {
+            // Retry for up to one second
+            bool create = mode != FileMode.Open;
+            for (int retries = 100; retries > 0; --retries) {
+                try {
+                    return new FileStream(file, mode, access, share);
+                } catch (FileNotFoundException) when (!create) {
+                    return null;
+                } catch (DirectoryNotFoundException) when (!create) {
+                    return null;
+                } catch (IOException) {
+                    if (create) {
+                        var dir = Path.GetDirectoryName(file);
+                        try {
+                            Directory.CreateDirectory(dir);
+                        } catch (IOException) {
+                            // Cannot create directory for DB, so just bail out
+                            return null;
+                        }
+                    }
+                    Thread.Sleep(10);
+                } catch (NotSupportedException) {
+                    return null;
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Normalizes and returns the provided path.
+        /// </summary>
+        /// <exception cref="ArgumentException">
+        /// If the provided path contains invalid characters.
+        /// </exception>
+        public static string NormalizePath(string path) {
```

Comment:
We have `Microsoft.PythonTools.Infrastructure.PathUtils.NormalizePath`. How this one is different?

Reply:
The Analysis project has no dependencies, so it can't access that.

We can change the existing one to redirect here and only have one implementation (I *think* everything that uses Common also uses Analysis, though it isn't required, and arguably these helpers should be private since the Analysis API is going to be public/stable), but I've mainly just been copy-pasting into the new namespace.